### PR TITLE
Update appConfig version to 1.1.2 for newer version of azure core dependency

### PR DESCRIPTION
**Pull request by mssfang**
Still waiting azure core http netty to release first. Then merge this.

### CODE REVIEW

File: eng/jacoco-test-coverage/pom.xml [hunk 1]
```diff
@@ -99,7 +99,7 @@
     <dependency>
       <groupId>com.azure</groupId>
       <artifactId>azure-data-appconfiguration</artifactId>
-      <version>1.2.0-beta.1</version> <!-- {x-version-update;com.azure:azure-data-appconfiguration;current} -->
+      <version>1.2.0</version> <!-- {x-version-update;com.azure:azure-data-appconfiguration;current} -->
```
Comment (alzimmermsft) [inline eng/jacoco-test-coverage/pom.xml:102]: I believe this could be a patch release of 1.1.2

**Comment by mssfang** [comment]
will update the changelog date if schedule changed.

**Comment by alzimmermsft** [comment]
You'll want to wait on `azure-identity` shipping and merging a version increment in before shipping.

**Pull request merged**